### PR TITLE
Update `pyppeteer` to `0.2.6`

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR updates `pyppeteer` to `0.2.6`. I tried to update the `.patch` files to reflect the code changes in the new version.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ajschmidt8** [comment]
@conda-forge-admin, please rerender

**Comment by bollwyvl** [comment]
Unless @kiwi0fruit thinks otherwise, I don't think the `downloader.py` patch is even required anymore, as it no longer uses an insecure-by-default settings... the code is still there, but commented out:

https://github.com/pyppeteer/pyppeteer/blame/0.2.6/pyppeteer/chromium_downloader.py#L76

File: recipe/0001-Relax-version-constraint-on-importlib-metadata.patch [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+diff --git a/pyppeteer-0.2.6/pyproject.toml b/pyppeteer-0.2.6/pyproject.toml
+index 31bb753..d3045a7 100644
+--- a/pyppeteer-0.2.6/pyproject.toml
++++ b/pyppeteer-0.2.6/pyproject.toml
+@@ -46,7 +46,7 @@ pyppeteer-install = 'pyppeteer.command:install'
+ [tool.poetry.dependencies]
+ python = "^3.6.1"
+ appdirs = "^1.4.3"
+-importlib-metadata = ">=1.4"
++importlib-metadata = { version = ">=2.1.1", python = "<3.8" }
```
Comment (bollwyvl) [inline recipe/0001-Relax-version-constraint-on-importlib-metadata.patch]: This has also been loosened, and we likely no longer need the patch:
https://github.com/pyppeteer/pyppeteer/blame/0.2.6/pyproject.toml#L49

File: recipe/conda-forge.patch [hunk 2]
```diff
@@ -1,35 +1,38 @@
...
- 
-         try:
-             total_length = int(data.headers['content-length'])
--- 
-2.28.0
-
+-        r = http.request('GET', url, preload_content=False)
++        r = https.request('GET', url, preload_content=False)
+         if r.status >= 400:
+             raise OSError(f'Chromium downloadable not found at {url}: ' f'Received {r.data.decode()}.\n')
```
Comment (bollwyvl) [inline recipe/conda-forge.patch]: This also looks to be upstreamed:

https://github.com/pyppeteer/pyppeteer/blame/0.2.6/pyppeteer/chromium_downloader.py#L83

**Comment by kiwi0fruit** [comment]
The download doesn't seem to be secure in https://github.com/pyppeteer/pyppeteer/blob/0.2.6/pyppeteer/chromium_downloader.py#L78

Previously this code was insecure but gave warnings. I bet it's still insecure. May be urllib3 started using sertifi by default? I doubt that's the case but may be? Worth checking. It also worth asking pyppeteer developers directly why they still do not use sertifi download. I bet they don't know about relevant PR.

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
> I bet it's still insecure. 

Yep, the `urllib3` docs still claim to require the `ca_certs` kwarg. This was merged, once:

https://github.com/pyppeteer/pyppeteer/pull/76/files

But apparently never made it to the release branch. :facepalm: 

I reckon we re-build the patch, adding just the ssl kwargs. Sigh.

**Comment by bollwyvl** [comment]
Ugh, the `.tar.gz` has windows line endings.

File: recipe/0001-use-certifi.patch [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+-    # download without certificate verification. This is *strongly* as it
+-    # opens the code to man-in-the-middle (and other) vulnerabilities; see
+-    # https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
+-    # for more.
+-    # urllib3.disable_warnings()
+-
+-    with urllib3.PoolManager() as http:
++    import certifi
++
++    with urllib3.PoolManager(cert_reqs='CERT_REQUIRED', ca_certs=certifi.where()) as http:
```
Comment (bollwyvl) [inline recipe/0001-use-certifi.patch:19]: @kiwi0fruit look ok?

**Comment by kiwi0fruit** [comment]
@bollwyvl Somebody should re-introduce this patch to the pyppeteer new repo. I guess they forgot about this problem. Unfortunately I won't open there an issue or make a PR at the moment. I don't know when I would have motivation for this. If you have time and motivation may be you can do this?

File: recipe/0001-use-certifi.patch (hunk 3 above)
Comment (kiwi0fruit) [reply recipe/0001-use-certifi.patch:19]: I don't know if this patch works (modifies the file in a right way) but that's the gist of the modification to make in it. May be we need regex patch here too like in the pyee?

Comment (bollwyvl) [reply recipe/0001-use-certifi.patch:19]: >  May be we need regex patch here too like in the pyee?

My memory fails me: what patch would that be? Generally, the bog-standard `.patch` files, applied well before any scripts start hacking, seem to be more effective over time, even though they "break" more frequently (that's a good thing, in this case).

**Comment by bollwyvl** [comment]
> @bollwyvl Somebody should re-introduce this patch to the pyppeteer new repo. I guess they forgot about this problem. Unfortunately I won't open there an issue or make a PR at the moment. I don't know when I would have motivation for this. If you have time and motivation may be you can do this?

https://github.com/pyppeteer/pyppeteer/pull/301

**Comment by ajschmidt8** [comment]
> > @bollwyvl Somebody should re-introduce this patch to the pyppeteer new repo. I guess they forgot about this problem. Unfortunately I won't open there an issue or make a PR at the moment. I don't know when I would have motivation for this. If you have time and motivation may be you can do this?
> 
> [pyppeteer/pyppeteer#301](https://github.com/pyppeteer/pyppeteer/pull/301)

@bollwyvl, this patch will presumably go out in a new release (i.e. `0.2.7`). For now, can we merge this PR to get `0.2.6` published?

**Comment by bollwyvl** [comment]
@ajschmidt8 i'm _relatively_ confident in this, as it has added a test of the patched behavior, but would be more comfortable if @kiwi0fruit could toss an approval... we can rejigger the recipe to make the tarballs available, if that would help.

**Comment by ajschmidt8** [comment]
> we can rejigger the recipe to make the tarballs available, if that would help.

@bollwyvl, not sure I understand what you mean by this. Ultimately I'm just looking for `0.2.6` to be published to anaconda.org. Can this PR be merged yet?

**Comment by SylvainCorlay** [comment]
Thanks for pushing the certifi patch again @bollwyvl!

So your patch was integrated upstream, and the rest of this PR seems quite standard (removing upstreamed patches and updating versions), I htink it is fine to go ahead and merge it.

**Comment by SylvainCorlay** [comment]
:tada:

**Pull request merged**